### ISSUE TITLE
Eta reduction under `where` 

### ISSUE THREAD

**Issue by ChefYeum** [ndmitchell/hlint#1264]
Eta reduction is not suggested under `where` clause.

It *is* suggested in the following line:
```hs
someFunc x y z = someOtherFunc x y z
```

But not when you put the same line under `where` clause:
```hs
f a b = someOperations $ someFunc a b c
  where someFunc x y z = someOtherFunc x y z
```

**Comment by ChefYeum** [comment]
Is it related to #1082?

**Comment by ndmitchell** [comment]
I don't think its related to #1082, but no real idea why it doesn't fire - I would have expected it to do so. Someone will need to investigate.

**Comment by zliu41** [comment]
It's because eta reduce is triggered on `LHsDecl`s, and there's only one `LHsDecl` - the top level function. The local function under `where` is a `LHsBind`.

I agree that it makes sense to fire eta reduce in this case, so perhaps eta reduce should be changed to run on all `LHsBind`s?

**Comment by ndmitchell** [comment]
That makes sense - PR welcome.

**Comment by zliu41** [comment]
Will put up a PR tomorrow!
